_rules/heading-accessible-name-is-descriptive-fdaadc.md

```diff
@@ -0,0 +1,290 @@
+---
+id: fdaadc
+name: Heading accessible name is descriptive
+rule_type: atomic
+
+description: |
+   This rule checks that the accessible name of a heading describes the topic or purpose of the content.
+   
+accessibility_requirements:
+- wcag20:2.4.6 # Headings and labels (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+
+input_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Anne Thyme Nørregaard
+- Dagfinn Rømen
+- Geir Sindre Fossøy
+- Carlos Duarte
+---
+
+## Applicability
+
+This rule applies to any element with the [semantic role](#semantic-role) of `heading` that is [included in the accessibility tree](#included-in-the-accessibility-tree), has an [accessible name](#accessible-name) that is not only [whitespace](#whitespace), and where:
+- the [accessible name](#accessible-name) does not [match](#matching-characters) the [visible](#visible) heading that is the combined [characters](#character) formed by the [visible](#visible) [descendant text content](#) and [characters](#character) in [images of text](https://www.w3.org/TR/WCAG21/#dfn-images-of-text), OR
```

Comment:
[visible](#visible) [descendant text content](#) --> [visible](#visible) [shadow-including] [descendant text content](#) - except that the descendant text content definition doesn't work in this case

images of text also needs to be visible shadow-including descendants

Reply:
flat tree text content: the concatenation of the data of all text node descendants of node in the flat tree, in tree order.